fixtures/entrypoints-rpc-tests/tests/entrypoints.spec.ts

```diff
@@ -0,0 +1,825 @@
+			export default {
+				async fetch(request, env, ctx) {
+					const response = await env.SERVICE.fetch("https://placeholder:9999/", {
+						method: "POST",
+						cf: { thing: true },
+					});
+					const text = await response.text();
+					const pong = await env.SERVICE.ping();
+					return new Response(\`\${text} \${pong}\`);
+				}
+			}
+		`,
+	});
+
+	await waitFor(async () => {
+		const response = await fetch(url);
+		const text = await response.text();
+		// Check protocol, host, and cf preserved
+		expect(text).toBe('POST https://placeholder:9999/ {"thing":true} pong');
+	});
+});
+
+test("should support named entrypoints in pages dev", async ({ dev }) => {
+	await dev({
+		"wrangler.toml": dedent`
+			name = "bound"
+			main = "index.ts"
+		`,
+		"index.ts": dedent`
+			import { WorkerEntrypoint } from "cloudflare:workers";
+			export class ThingEntrypoint extends WorkerEntrypoint {
+				ping() {
+					return "pong";
+				}
+			};
+			export default {}; // Required to treat as modules format worker
+		`,
+	});
+
+	const files = {
+		"functions/index.ts": dedent`
+			export const onRequest = async ({ env }) => {
+				return new Response(await env.SERVICE.ping());
+			};
+		`,
+	};
+	const { url } = await dev(
+		files,
+		["--compatibility-flags=rpc", "--service=SERVICE=bound#ThingEntrypoint"],
+		/* pagesPublicPath */ "dist"
+	);
+
+	await waitFor(async () => {
+		const response = await fetch(url);
+		const text = await response.text();
+		expect(text).toBe("pong");
+	});
+});
+
+test("should support co-dependent services", async ({ dev }) => {
+	const { url } = await dev({
+		"wrangler.toml": dedent`
+			name = "a"
+			main = "index.ts"
+			compatibility_flags = ["rpc"]
+
+			[[services]]
+			binding = "SERVICE_B"
+			service = "b"
+			entrypoint = "BEntrypoint"
+		`,
+		"index.ts": dedent`
+			import { WorkerEntrypoint } from "cloudflare:workers";
+			export class AEntrypoint extends WorkerEntrypoint {
+				ping() {
+					return "a:pong";
+				}
+			};
+			export default {
+				async fetch(request, env, ctx) {
+					return new Response(await env.SERVICE_B.ping());
+				}
+			};
+		`,
+	});
+
+	await dev({
+		"wrangler.toml": dedent`
+			name = "b"
+			main = "index.ts"
+			compatibility_flags = ["rpc"]
+
+			[[services]]
+			binding = "SERVICE_A"
+			service = "a"
+			entrypoint = "AEntrypoint"
+		`,
+		"index.ts": dedent`
+			import { WorkerEntrypoint } from "cloudflare:workers";
+			export class BEntrypoint extends WorkerEntrypoint {
+				async ping() {
+					const result = await this.env.SERVICE_A.ping();
+					return \`b:\${result}\`;
+				}
+			};
+			export default {}; // Required to treat as modules format worker
+		`,
+	});
+
+	await waitFor(async () => {
+		const response = await fetch(url);
+		const text = await response.text();
+		expect(text).toBe("b:a:pong");
+	});
+});
+
+test("should support binding to Durable Object in another worker", async ({
+	dev,
+}) => {
+	// RPC isn't supported in this case yet :(
```

Comment:
Is that a runtime limitation? Or a local dev thing?

Reply:
Do we want to add an expect-error test? (would be good to assert we have a good error message in this case if possible)